app/web/src/helpers/cadPackages/jsCad/jsCadController.ts

```diff
@@ -70,15 +72,44 @@ function CSG2Object3D(obj) {
 }
 
 let scriptWorker
-let currentParameters = {}
-const scriptUrl = '/demo-worker.js'
-let resolveReference = null
-let response = null
 
-const callResolve = () => {
-  if (resolveReference) resolveReference()
-  resolveReference
+type ResolveFn = (RenderResponse) => void
+
+class WorkerHelper {
+  callResolve: null | ResolveFn = null
+  previousCode = ''
+  resolver = (arg?: RenderResponse) => {
+    this.callResolve && this.callResolve(arg)
+    this.callResolve = null
+  }
+  render = (
```

Comment:
fundamentally i haven't changed how your code works, I just wanted to group things a little better in this file as I found some of the module level variables hard to keep track of and so instead moved some of the stuff that required state into this class. Plus one other thing that made me uncomfortable was how the function awaited a promise just to return another module level variable `response`.

This render function now sets up promise -> posts a message to the worker -> then returns the promise where the promise now contains the final response. I at least personally find it easier to reason about.

Reply:
yes, it looks cleaner. thnx :)